### PR TITLE
Add finder method and appeal model spec

### PR DESCRIPTION
**Pull request by kahlouie**
Connects #5626

### CODE REVIEW

File: spec/factories/appeal.rb [hunk 1]
```diff
@@ -11,5 +11,9 @@
         "783740847"
       end
     end
+
+    uuid do
+      "a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11"
```
Comment (shanear) [inline spec/factories/appeal.rb]: I think we may want this to generate a random UUID, since we will be using these factories for setting up demo data too

File: spec/models/appeal_spec.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+          uuid: "a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11"
+        )
+      end
+
+      it "finds the appeal" do
+        expect(Appeal.find_appeal_or_legacy_appeal_by_id(appeal.uuid)).to eq(appeal)
+      end
+
+      it "return nil for a non-existant one" do
+        expect(Appeal.find_appeal_or_legacy_appeal_by_id("11111111-aaaa-bbbb-CCCC-999999999999")).to be_nil
```
Comment (shanear) [inline spec/models/appeal_spec.rb]: Not sure what we want here, `nil`, or to raise `ActiveRecord::RecordNotFound` which is the default functionality of `.find`

@mdbenjam got opinions here?

File: app/models/appeal.rb [hunk 3]
```diff
@@ -4,6 +4,15 @@ class Appeal < ApplicationRecord
 
   has_many :request_issues, as: :review_request
 
+  def self.find_appeal_or_legacy_appeal_by_id(id)
+    uuid_regex = /^\h{8}-\h{4}-\h{4}-\h{4}-\h{12}$/
```
Comment (shanear) [inline app/models/appeal.rb]: I feel like this might belong as a constant somewhere. I just don't know where...

File: spec/models/appeal_spec.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+          vacols_id: "1234567",
+          vbms_id: "111223333S",
+          type: "Original",
+          decision_date: 365.days.ago.to_date,
+          issues: [
+            Generators::Issue.build(codes: %w[02 01]),
+            Generators::Issue.build(codes: %w[02 02]),
+            Generators::Issue.build(codes: %w[02 03])
+          ]
+        )
```
Comment (shanear) [inline spec/models/appeal_spec.rb]: I think the data setup here may be changing with https://github.com/department-of-veterans-affairs/caseflow/pull/5679

So we will need to change this if that PR is merged.

File: spec/models/appeal_spec.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      end
+
+      it "finds the appeal" do
+        legacy_appeal.save
+        expect(Appeal.find_appeal_or_legacy_appeal_by_id(veteran_file_number)).to
+        eq([legacy_appeal])
```
Comment (shanear) [inline spec/models/appeal_spec.rb]: why does this return an array? I'm thinking it shouldn't

File: spec/factories/appeal.rb (hunk 1 above)
Comment (mdbenjam) [reply spec/factories/appeal.rb]: Interesting, is uuid not the primary key on the table? It seems like if this is the primary key then won't FactoryBot/Rails/Postgres auto generate it properly?

If we want it as a separate field, then it probably should be a random UUID, otherwise we won't be able to have multiple appeals in our tests.

File: spec/models/appeal_spec.rb (hunk 2 above)
Comment (mdbenjam) [reply spec/models/appeal_spec.rb]: I agree with raising `ActiveRecord::RecordNotFound`

File: app/models/appeal.rb [hunk 6]
```diff
@@ -4,6 +4,15 @@ class Appeal < ApplicationRecord
...
+    if uuid_regex.match(id)
+      find_by_uuid(id)
+    else
+      LegacyAppeal.fetch_appeals_by_file_number(id)
```
Comment (mdbenjam) [inline app/models/appeal.rb]: I think we need to look for legacy appeals id, not file number. We also will need another method to `find_appeal_by_id_or_legacy_appeal_by_vacols_id`. I'm happy to make that in another PR when we fix Reader to work with this.

File: spec/models/appeal_spec.rb (hunk 5 above)
Comment (mdbenjam) [reply spec/models/appeal_spec.rb]: Agreed, there should only be one appeal per id.
